federationapi/routing/send.go

```diff
@@ -135,22 +124,28 @@ func (t *txnReq) processTransaction() (*gomatrixserverlib.RespSend, error) {
 		}
 		if err := json.Unmarshal(pdu, &header); err != nil {
 			util.GetLogger(t.context).WithError(err).Warn("Transaction: Failed to extract room ID from event")
-			return nil, unmarshalError{err}
+			continue
```

Comment:
We should probably just add an error to the `results` map like we do elsewhere (in which case just factor out the for loop to a new function which returns an `error` and set that if it's non-nil.

Reply:
The difficulty here is that we don't know the event ID until after `gomatrixserverlib.NewEventFromUntrustedJSON`.

Reply:
oh hah. Fair enough, might be worth adding a quick comment about that.